File: Agora-Signaling-Tutorial-iOS-Objective-C/Agora-Signaling-Tutorial/Utils/LogWriter.m
//
//  LogWriter.m
//  Agora-Signaling-Tutorial
//
//  Created by Zhangji on 2018/11/16.
//  Copyright © 2018 Agora. All rights reserved.
//

#import "LogWriter.h"

@implementation LogWriter
+ (NSURL *)documentDir {
    return [[NSFileManager defaultManager] URLForDirectory:NSDocumentDirectory inDomain:NSUserDomainMask appropriateForURL:nil create:NO error:nil];
}

+ (NSURL *)url {
    return [[self documentDir] URLByAppendingPathComponent:@"log.txt"];
}

+ (void)initLogWriter {
    NSMutableData *data = [[NSMutableData alloc] init];
    [data appendData:[@"" dataUsingEncoding:NSUTF8StringEncoding allowLossyConversion:YES]];
    [data writeToURL:[self url] atomically:YES];
}

+ (void)writeLog:(NSString *)log {
    NSString *logs = [NSString stringWithFormat:@"%@\n", log];
    NSData *appendedData = [logs dataUsingEncoding:NSUTF8StringEncoding allowLossyConversion:YES];
    NSFileHandle *writeHandler = [NSFileHandle fileHandleForWritingToURL:[self url] error:nil];
    [writeHandler seekToEndOfFile];
    [writeHandler writeData:appendedData];
}
@end

File: Agora-Signaling-Tutorial-iOS-Swift/Agora-Signaling-Tutorial/Utils/LogWriter.swift
//
//  LogWriter.swift
//  Agora-Signaling-Tutorial
//
//  Created by ZhangJi on 04/12/2017.
//  Copyright © 2017 ZhangJi. All rights reserved.
//

import UIKit

var documentDir: URL! {
    return try! FileManager.default.url(for: .documentDirectory, in: .userDomainMask, appropriateFor: nil, create: false)
}
var url: URL! {
    return documentDir.appendingPathComponent("log.txt")
}

class LogWriter: NSObject {

    static func initLogWriter() {
        let data = NSMutableData()
        data.append("".data(using: String.Encoding.utf8, allowLossyConversion: true)!)
        data.write(to: url, atomically: true)
    }
    
    static func write(log: String) {
        let logs = log + "\n"
        let appendedData = logs.data(using: String.Encoding.utf8, allowLossyConversion: true)!
        let writeHandler = try? FileHandle(forWritingTo:url)
        writeHandler!.seekToEndOfFile()
        writeHandler!.write(appendedData)
    }
}
